File: Problem146.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include "util.c"

const unsigned long long N = 150000000;

int main(int argc, char **argv) {
    unsigned long long sum = 0;

    for (unsigned long long x = 2; x < N; x += 2) {
        if (x % 3 && x % 7 && x % 13) {
            unsigned long long x2 = x * x;
            if (is_prime(x2 + 1)
             && is_prime(x2 + 3)
             && is_prime(x2 + 7) 
             && is_prime(x2 + 9) 
             && is_prime(x2 + 13) 
             && is_prime(x2 + 27)
             && !is_prime(x2 + 5)
             && !is_prime(x2 + 11)
             && !is_prime(x2 + 15)
             && !is_prime(x2 + 17)
             && !is_prime(x2 + 19)
             && !is_prime(x2 + 21)
             && !is_prime(x2 + 23)
             && !is_prime(x2 + 25)) {
                sum += x;
            }
        }
        if (x % 1000000 == 0) {
            printf("finished %llu\n", x);
        }
    }
    printf("%llu\n", sum);
}

File: Problem146.py
from util import *

N = 150000000
N = 1000000
sum = 0
for i in range(10, N, 2):
    if i % 1000000 == 0:
        print('finished %s' % (i))
    if i % 3 == 0:
        continue
    elif i % 7 == 0:
        continue
    elif i % 13 == 0:
        continue
    i2 = i * i
    if is_prime(i2 + 1) and is_prime(i2 + 3) and is_prime(i2 + 7) and is_prime(i2 + 9) and is_prime(i2 + 13) and is_prime(i2 + 27):
        print(i)
        sum += i
print(sum)